src/main/java/org/mifos/connector/bulk/camel/routes/FileProcessingRoute.java

```diff
@@ -0,0 +1,110 @@
+package org.mifos.connector.bulk.camel.routes;
+
+import com.fasterxml.jackson.databind.MappingIterator;
+import com.fasterxml.jackson.databind.SequenceWriter;
+import com.fasterxml.jackson.dataformat.csv.CsvMapper;
+import com.fasterxml.jackson.dataformat.csv.CsvSchema;
+import org.mifos.connector.bulk.schema.Transaction;
+import org.mifos.connector.bulk.schema.TransactionResult;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.mifos.connector.bulk.zeebe.ZeebeVariables.FILE_NAME;
+
+@Component
+public class FileProcessingRoute extends BaseRouteBuilder {
+
+    private static final String TRANSACTION_LIST = "transactionList";
+    private static final String TRANSACTION_LIST_LENGTH = "transactionListLength";
+    private static final String TOTAL_AMOUNT = "totalAmount";
+    private static final String ONGOING_AMOUNT = "ongoingAmount";
+    private static final String FAILED_AMOUNT = "failedAmount";
+    private static final String COMPLETED_AMOUNT = "completedAmount";
+    private static final String LOCAL_FILE_PATH = "localFilePath";
+    private static final String RESULT_TRANSACTION_LIST = "resultTransactionList";
+    private static final String OVERRIDE_HEADER = "overrideHeader";
+
+    @Autowired
+    private CsvMapper csvMapper;
+
+    @Override
+    public void configure() throws Exception {
+
+        /**
+         * Parse the [Transaction] array from the csv file
+         * exchangeInput: [LOCAL_FILE_PATH] the absolute path to the csv file
+         * exchangeOutput: [TRANSACTION_LIST] containing the list of [Transaction]
+         */
+        from("direct:get-transaction-array")
+                .id("direct:get-transaction-array")
+                .log("Starting route direct:get-transaction-array")
+                .process(exchange -> {
+                    Double totalAmount = 0.0;
+                    Long failedAmount = 0L;
```

Comment:
why the amount fields are handled as Long? it should be double or Bigdecimal

Reply:
This file was copied from another service called bulk processor. I have made the changes to BigDecimal.